### DISCUSSION TITLE
Why the "Action items" panel doesn't show all warnings in the current file?

### DISCUSSION THREAD

**Discussion by negora** [apache/netbeans#4764] [Q&A]
Hi:

There is something in NetBeans IDE that has always intrigued me: Why the *Action items* panel doesn't show all warnings in the current file? By *current file* I mean the file that is currently being shown in the editor.

I understand that NB doesn't show the warnings of a whole project, because that would be really expensive. But for a file that is already open, the analysis is already done, Right? The *Action items* panel even has a button to show only the items related to the file being edited...

What's the reason behind this issue? Is it so by design? Is it because of some kind of internal limitation? Or it's just that most people have enough with the stripes in the right-side of the editor?

Thank you!

**Comment by Chris2011** [comment]
In general it should work, if it is not working, we need more information from your side to identify the problem.

**Comment by negora** [comment]
Hi @Chris2011. Thanks for answering.

It's very easy to demonstrate it. Consider this code:

```java
public class MyClass {

    // TODO: pending task

    private void test (int x) {
    }

}
```

The editor shows 2 warnings:

* *Variable x is never read*
* *test is never used*

![image](https://user-images.githubusercontent.com/2098654/195010333-bea82cf6-f662-4e3a-a7fc-81a4830ce42c.png)

However, the *Action items* panel doesn't show any warning, only the comment about the pending task:

![image](https://user-images.githubusercontent.com/2098654/195010680-1ead723c-2e6a-4486-99dc-cb09f2339e12.png)

I believe that this problem is not something new or a regression. I don't remember when I started to use the panel, but I think that it has always behaved like this: It shows errors, to-do tasks and only a few specific warnings about Java files (ie: some about serialization problems). It also shows warnings related to JS and HTML, but here I'm just referring to Java.

By the way, I've tested this using clean configurations in versions 13, 14 and 15. I could try it from version 3.6 onward (I store all the binaries) but that I don't remember if this panel even existed (I didn't use it back then).

**Reply by Chris2011** [reply]
Reproducable, thanks for the info. I dunno when this was broken.

**Comment by mbien** [comment]
yeah I noticed this too. Never had time to look into it. Would be good to repair this again. I believe the only item which works reliably are TODOs - so there is a start ;)

**Comment by negora** [comment]
@mbien @Chris2011 Thanks for answering.

In case that it's of any help, here there are 2 type of warnings that the *Action items* panel does show. Look at this code:

```java
public final class Artist
		implements Serializable {

	private Set<String> nicknames;

}
```

This code triggers 3 warnings in the editor:

* *[serial] serializable class com.negora.app.Artist has no definition of serialVersionUID*
* *[serial] non-transient instance field of a serializable class declared with a non-serializable type*
* *Variable nicknames is never read*

Of all these 3 warnings, only the 2 related to serialization are shown by the panel:

![image](https://user-images.githubusercontent.com/2098654/195388599-0db19613-bad5-4a20-a1cb-5340dbc2536e.png)

Maybe the code that shows these warnings about serialization issues has the clue to fix the other warnings?

By the way, these warnings are also shown when you expand the scope of the panel to the current project and to all projects, not only for the current file. I don't know whether that's important. But just in case... 

Should this discussion be converted into a proper issue? Do you want me to open one? Thank you!

**Reply by Chris2011** [reply]
So if it is not working anymore as it was before then it is a bug and should be reported as one, yes. And yes it seems broken. But maybe we should test a bit more to get specific steps to reproduce. But in your simple test I could reproduce it. So it seems enough. Just in case someone else could not reproduce. Also all environment versions are needed (operating system, jdk and which version, etc. as usual)

**Reply by negora** [reply]
@Chris2011 OK. Thank you. I'll open a proper issue as soon as possible.

**Comment by negora** [comment]
I've created the issue #4790. Thank you for your guidance.
